objects/src/notes/output.rs

```diff
@@ -0,0 +1,99 @@
+use miden_crypto::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{NoteConsumable, NoteEnvelope, NoteId, NoteMetadata, NoteVault};
+use crate::Digest;
+
+const PUBLIC: u8 = 0;
+const PRIVATE: u8 = 1;
+
+/// The types of note outputs supported by the transaction kernel.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum NoteOutput {
+    Public(NoteConsumable),
+    Private(NoteEnvelope),
+}
```

Comment:
Another thing to consider in a future PR: I'm not sure the variant names here is what we want to have long term. Specifically, we may know full details of a private note, and it would make sense to be able to keep track of that. So, maybe the variant names should be something like:

```Rust
pub enum OutputNote {
    Full(Note),
    Minimal(NoteEnvelope), // not a great name but couldn't come up with anything better
}
```

Let's create an issue to come back to this later.

Reply:
Let's create an issue for refactoring the `OutputNote` along the lines of what I described above.